spec/octopus/relation_proxy_spec.rb

```diff
@@ -28,6 +28,28 @@
       expect(not_relation.current_shard).to eq(@relation.current_shard)
     end
 
+    context 'when a new relation is constructed from the original relation' do
+      context 'and a where(...) is used' do
+        it 'does not tamper with the original relation' do
+          relation = Item.using(:canada).where(id: 1)
+          original_sql = relation.to_sql
+          new_relation = relation.where(id: 2)
+          expect(relation.to_sql).to eq(original_sql)
+          puts new_relation.to_sql
+        end
+      end
+
+      context 'and a where.not(...) is used' do
```

Comment:
This was never broken b/c `.where` returns a `ActiveRecord::QueryMethods::WhereChain`, which returns `false` for `respond_to?(:all)`.